framework/include/materials/MaterialProperty.h

```diff
@@ -180,6 +182,21 @@ class MaterialPropertyBase : public PropertyValue
    */
   virtual void load(std::istream & stream) override;
 
+  void name(const MaterialPropertyName & name_in)
```

Comment:
typically `name()` suggested a getter but this is a setter, so suggest to rename it to `setName`.